src/codegen/revert.rs

```diff
@@ -57,11 +64,20 @@ impl SolidityError {
     }
 
     /// Return the selector of the error.
-    pub fn selector(&self) -> u32 {
+    pub fn selector(&self, ns: &Namespace) -> u32 {
         match self {
             Self::Empty => unreachable!("empty return data has no selector"),
             Self::String(_) => ERROR_SELECTOR,
             Self::Panic(_) => PANIC_SELECTOR,
+            Self::Custom { error_no, .. } => {
+                let mut buf = [0u8; 32];
+                let mut hasher = Keccak::v256();
+                let signature =
+                    ns.signature(&ns.errors[*error_no].name, &ns.errors[*error_no].fields);
+                hasher.update(signature.as_bytes());
+                hasher.finalize(&mut buf);
+                u32::from_be_bytes([buf[0], buf[1], buf[2], buf[3]])
```

Comment:
What about `&buf[0..4]`?

Reply:
`u32::from_be_bytes` needs a `[u8; 4]` and not a `&[u8]`